marimo/_plugins/ui/_impl/tables/narwhals_table.py

```diff
@@ -0,0 +1,276 @@
+        return NarwhalsTableManager(self.data[offset:count])
+
+    def search(self, query: str) -> TableManager[Any]:
+        query = query.lower()
+
+        expressions: list[Any] = []
+        for column in self.data.columns:
+            dtype = self.data[column].dtype
+            if dtype == nw.String:
+                expressions.append(nw.col(column).str.contains(query))
+            # TODO: Unsupported by narwhals
+            # elif dtype == nw.List(nw.String):
+            #     expressions.append(
+            #         nw.col(column).cast(nw.String).str.contains(query)
+            #     )
+            elif (
+                dtype.is_numeric()
+                or is_narwhals_temporal_type(dtype)
+                or dtype == nw.Boolean
+            ):
+                expressions.append(
+                    nw.col(column).cast(nw.String).str.contains(f"(?i){query}")
+                )
+
+        if not expressions:
+            return NarwhalsTableManager(self.data.filter(nw.lit(False)))
+
+        or_expr = expressions[0]
+        for expr in expressions[1:]:
+            or_expr = or_expr | expr
+
+        filtered = self.data.filter(or_expr)
+        return NarwhalsTableManager(filtered)
+
+    def get_summary(self, column: str) -> ColumnSummary:
+        summary = self._get_summary_internal(column)
+        for key, value in summary.__dict__.items():
+            if value is not None:
+                summary.__dict__[key] = _maybe_convert_as_py(value)
+        return summary
+
+    def _get_summary_internal(self, column: str) -> ColumnSummary:
+        # If column is not in the dataframe, return an empty summary
+        if column not in self.data.columns:
+            return ColumnSummary()
+        col = self.data[column]
+        total = len(col)
+        if is_narwhals_string_type(col.dtype):
+            return ColumnSummary(
+                total=total,
+                nulls=col.null_count(),
+                unique=col.n_unique(),
+            )
+        if col.dtype == nw.Boolean:
+            return ColumnSummary(
+                total=total,
+                nulls=col.null_count(),
+                true=cast(int, col.sum()),
+                false=cast(int, total - col.sum()),
+            )
+        if col.dtype == nw.Date:
+            return ColumnSummary(
+                total=total,
+                nulls=col.null_count(),
+                min=col.min(),
+                max=col.max(),
+                mean=col.mean(),
+                # TODO: Implement median not in narwhals
+                # median=col.median(),
+            )
+        if is_narwhals_temporal_type(col.dtype):
+            return ColumnSummary(
+                total=total,
+                nulls=col.null_count(),
+                min=col.min(),
+                max=col.max(),
+                mean=col.mean(),
+                # TODO: Implement median not in narwhals
+                # median=col.median(),
```

Comment:
For median, `col.quantile(0.50, interpolation="nearest")`?

Reply:
sorry already merged, will do this in my followup, if they don't release median by that time